scanner/scanner.go

```diff
@@ -316,8 +322,14 @@ func (s *Scanner) scanDoubleQuote(ctx *Context) (tk *token.Token, pos int) {
 			continue
 		} else if c == '\\' {
 			isFirstLineChar = false
-			if idx+1 < size {
-				nextChar := src[idx+1]
+			if nextChar != 0 {
+				// we need to treat this as a newline immediately and ignore the \ character
```

Comment:
It would be great if you could provide a comment in the source code with an example of why it is needed

Reply:
Fair point, I don't remember so I'll play with the tests to see if I can reproduce the original issue and provide context.

Reply:
hmm. I don't know what I was smoking at the time, this entire change is incorrect. The code does what it needs to do, will only add the relevant tests.